soh/soh/Enhancements/randomizer/randomizer_check_objects.cpp

```diff
@@ -0,0 +1,815 @@
+    { RC_ZD_GOSSIP_STONE, {RC_ZD_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_ZORAS_DOMAIN, "Gossip Stone", "ZD Gossip Stone"}},
+    { RC_ZR_NEAR_DOMAIN_GOSSIP_STONE, {RC_ZR_NEAR_DOMAIN_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_ZORAS_RIVER, "Near Domain Gossip Stone", "ZR Near Domain Gossip Stone"}},
+    { RC_ZR_NEAR_GROTTOS_GOSSIP_STONE, {RC_ZR_NEAR_GROTTOS_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_ZORAS_RIVER, "Near Grottos Gossip Stone", "ZR Near Grottos Gossip Stone"}},
+    { RC_ZR_OPEN_GROTTO_GOSSIP_STONE, {RC_ZR_OPEN_GROTTO_GOSSIP_STONE, RCVORMQ_VANILLA, RCTYPE_GOSSIP_STONE, RCAREA_ZORAS_RIVER, "Open Grotto Gossip Stone", "ZR Open Grotto Gossip Stone"}},
+};
+
+std::map<RandomizerCheckArea, std::string> rcAreas = {
+   { RCAREA_KOKIRI_FOREST, "Kokiri Forest"},
+   { RCAREA_LOST_WOODS, "Lost Woods"},
+   { RCAREA_SACRED_FOREST_MEADOW, "Sacred Forest Meadow"},
+   { RCAREA_HYRULE_FIELD, "Hyrule Field"},
+   { RCAREA_LAKE_HYLIA, "Lake Hylia"},
+   { RCAREA_GERUDO_VALLEY, "Gerudo Valley"},
+   { RCAREA_GERUDO_FORTRESS, "Gerudo Fortress"},
+   { RCAREA_WASTELAND, "Desert Wasteland"},
+   { RCAREA_DESERT_COLOSSUS, "Desert Colossus"},
+   { RCAREA_MARKET, "Hyrule Market"},
+   { RCAREA_HYRULE_CASTLE, "Hyrule Castle"},
+   { RCAREA_KAKARIKO_VILLAGE, "Kakariko Village"},
+   { RCAREA_GRAVEYARD, "Graveyard"},
+   { RCAREA_DEATH_MOUNTAIN_TRAIL, "Death Mountain Trail"},
+   { RCAREA_GORON_CITY, "Goron City"},
+   { RCAREA_DEATH_MOUNTAIN_CRATER, "Death Mountain Crater"},
+   { RCAREA_ZORAS_RIVER, "Zora's River"},
+   { RCAREA_ZORAS_DOMAIN, "Zora's Domain"},
+   { RCAREA_ZORAS_FOUNTAIN, "Zora's Fountain"},
+   { RCAREA_LON_LON_RANCH, "Lon Lon Ranch"},
+   { RCAREA_DEKU_TREE, "Deku Tree"},
+   { RCAREA_DODONGOS_CAVERN, "Dodongo's Cavern"},
+   { RCAREA_JABU_JABUS_BELLY, "Jabu Jabu's Belly"},
+   { RCAREA_FOREST_TEMPLE, "Forest Temple"},
+   { RCAREA_FIRE_TEMPLE, "Fire Temple"},
+   { RCAREA_WATER_TEMPLE, "Water Temple"},
+   { RCAREA_SPIRIT_TEMPLE, "Spirit Temple"},
+   { RCAREA_SHADOW_TEMPLE, "Shadow Temple"},
+   { RCAREA_BOTTOM_OF_THE_WELL, "Bottom of the Well"},
+   { RCAREA_ICE_CAVERN, "Ice Cavern"},
+   { RCAREA_GERUDO_TRAINING_GROUND, "Gerudo Training Grounds"},
+   { RCAREA_GANONS_CASTLE, "Ganon's Castle"},
+ };
+
+bool RandomizerCheckObjects::AreaIsDungeon(RandomizerCheckArea area) {
+    return area == RCAREA_GANONS_CASTLE ||
+           area == RCAREA_GERUDO_TRAINING_GROUND ||
+           area == RCAREA_ICE_CAVERN ||
+           area == RCAREA_BOTTOM_OF_THE_WELL ||
+           area == RCAREA_SHADOW_TEMPLE ||
+           area == RCAREA_SPIRIT_TEMPLE ||
+           area == RCAREA_WATER_TEMPLE ||
+           area == RCAREA_FIRE_TEMPLE ||
+           area == RCAREA_FOREST_TEMPLE ||
+           area == RCAREA_JABU_JABUS_BELLY ||
+           area == RCAREA_DODONGOS_CAVERN ||
+           area == RCAREA_DEKU_TREE;
```

Comment:
Not sure if it would have any performance impact, but this might be more readable as an array and `return std::find(std::begin(array), std::end(array), area) != std::end(array);`

Reply:
This is a pretty small and static set of checks though so maybe it's better to keep it as is.

Reply:
yeah, i feel like it's probably fine as is